account_invoice_delivered_amount/models/account_invoice_line.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    is_delivered = fields.Boolean(string="Deliverd", compute="_compute_delivered_amount", store=True)
+    deliverd_amount = fields.Monetary(string='Delivered Amount',
+        store=True, compute='_compute_delivered_amount')
+    
+    @api.depends("sale_line_ids", "product_id", "quantity")
```

Comment:
Listen to the specific field of sale.order.line.  We need to watch the delivered qty field.